hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/MasterQuotasObserver.java

```diff
@@ -65,8 +81,11 @@ public void postDeleteTable(
     }
     final Connection conn = ctx.getEnvironment().getConnection();
     Quotas quotas = QuotaUtil.getTableQuota(conn, tableName);
+    Quotas quotasAtNamespace = QuotaUtil.getNamespaceQuota(conn, tableName.getNamespaceAsString());
     if (quotas != null){
```

Comment:
Could consolidate these two branches, right?

Make this `if (quotas != null || quotasAtNamespace != null)` and then only call `removeTableSpaceLimit` when it is `quotas != null`.

Reply:
done